### PR TITLE
Git: check for empty values in received ReferencePointers

### PR DESCRIPTION
**Pull request by bzz**
This fixes current style-analyzer failure mode on staging:

```
{"app":"lookout","level":"debug","msg":"load trees for references: [{git://github.com/bblfsh/uast-viewer.git  07c123ff0bdbe7bd87f03c4d98f95d4ff2ec0df1}]","source":"git/service.go:103","time":"2018-08-24T18:01:04.635128964Z"}
{"app":"lookout","level":"info","msg":"fetching references for repository git://github.com/bblfsh/uast-viewer.git: [:]","source":"git/syncer.go:59","time":"2018-08-24T18:01:04.63538259Z"}
{"app":"lookout","duration":496267539,"error":"malformed refspec, separators are wrong","grpc.code":2,"grpc.method":"GetFiles","grpc.service":"pb.Data","grpc.start_time":"2018-08-24T18:01:04Z","level":"info","msg":"streaming server call finished","source":"grpchelper/logger.go:19","span.kind":"server","system":"grpc","time":"2018-08-24T18:01:05.131271264Z"}
```

It validates user input for `GetFiles` and `GetChanges` in the DataService, before passing it in to any of the LibraryCommitLoader or StorerCommitLoader.

Was not reproducible locally \w SDK+style-analyzer, as SDK binary only uses StorerCommitLoader that does not use Syncer to fetch.

Current behavior allows `GetFiles` request with empty ReferenceName in Revision, by explicitly [falling back to the default RefSpec](https://github.com/src-d/lookout/pull/193/files#diff-6bd7698193142bec9f5cd80af5f41970R53), before fetching.

`GetChanges` without ReferenceNames in Head or Base are prohibited as it's expected to be used on PRs.

Test plan:
 - `go test ./service/git`

### CODE REVIEW

File: cmd/lookout/event.go [hunk 1]
```diff
@@ -64,13 +64,13 @@ func (c *EventCommand) resolveRefs() (*lookout.ReferencePointer, *lookout.Refere
 
 	fromRef := lookout.ReferencePointer{
 		InternalRepositoryURL: "file://" + fullGitPath,
-		ReferenceName:         plumbing.ReferenceName(c.RevFrom),
+		ReferenceName:         plumbing.HEAD,
```
Comment (carlosms) [inline cmd/lookout/event.go]: Why is this change needed?

Comment (bzz) [reply cmd/lookout/event.go]: Because it's a bug otherwise - the purpose of this field to construct a [RefSpec](https://git-scm.com/book/en/v2/Git-Internals-The-Refspec), so it has to be reference, and not a commit hash as before.

The reason it did not fail at runtime when using SDK binary is that we never fetched. Lookout SDK, as opposed to serve, did not use LibraryCommitLoader, which before this change, was the only one way to validate it through actually trying to fetch a repo.

**Comment by bzz** [comment]
History re-written, waiting for CI to pass
